app/controllers/inventories_controller.rb

```diff
@@ -6,17 +6,50 @@
 
 class InventoriesController < ApplicationController
   before_action :require_can_view_projects!
-  before_action :require_can_edit_projects!, only: [:edit, :update]
-  before_action :set_inventory, only: [:show, :update, :edit]
-  before_action :set_project, only: [:edit, :update]
+  before_action :require_can_edit_projects!, only: [:new, :create, :edit, :update, :destroy]
+  before_action :set_inventory, only: [:show, :update, :edit, :destroy]
+  before_action :set_project, only: [:edit, :update, :destroy]
+  before_action :set_project_from_params, only: [:new, :create]
 
   include ArelHelper
 
+  def new
+    @inventory = @project.inventories.build
+    @fields = processed_form_fields
+    render layout: 'ajax_modal_rails/content'
+  end
+
+  def create
+    inventory = @project.inventories.create(
+      **manual_inventory_params,
+      manual_entry: true,
+      ProjectID: @project.ProjectID,
+      DateCreated: Time.current,
+      DateUpdated: Time.current,
+      UserID: current_user.email,
+    )
+    inventory.InventoryID = "m-#{inventory.id}"
+    inventory.ExportID = "m-#{inventory.id}"
+    inventory.save
+    redirect_to project_path(@project)
```

Comment:
Perhaps respond with?  That will set the flash message and re-load the form if there are errors.

Reply:
✅